### PR TITLE
feat(analytics): Analytics dashboard

### LINKED ISSUE #32724: Create Analytics Dashboard for dotCMS Admin Interface
## User Story

As a dotCMS administrator, I want to access a comprehensive analytics dashboard so that I can monitor site performance, user engagement, and content effectiveness in one centralized location.

## Acceptance Criteria

- \[ \] Create a new Analytics dashboard section in the dotCMS admin interface
- \[ \] Implement the dashboard layout according to the provided mock design
- \[ \] Display key analytics metrics \(page views, user sessions, content performance\)
- \[ \] Include interactive charts and visualizations
- \[ \] Integrate with existing dotCMS analytics data sources
- \[ \] Add proper loading states and error handling
- \[ \] Follow dotCMS UI/UX guidelines and design system

## Proposed Objective

Core Features

## Proposed Priority

Priority 2 - Important

## External Links

- Mock design attached

Image

- Related analytics API documentation \(if applicable\)

## Assumptions & Initiation Needs

- Analytics data collection is already implemented in dotCMS
- Backend APIs for analytics data are available or will be created
- Design system components for charts/graphs are available
- Access to the mock design specifications and requirements

## Quality Assurance Notes

- Test dashboard performance with large datasets
- Verify data accuracy against source analytics
- Test for proper error handling and edge cases

## Sub-Tasks & Estimates

- \[ \] Backend API endpoints for analytics data \(5 points\)
- \[ \] Frontend dashboard component structure \(3 points\)
- \[ \] Chart and visualization components \(8 points\)
- \[ \] Integration with existing dotCMS admin \(2 points\)
- \[ \] Testing and QA \(3 points\)
- \[ \] Documentation and user guide \(2 points\)

### PR DESCRIPTION
**Pull request by oidacra**
## Summary

  This PR introduces a comprehensive analytics dashboard feature that provides data visualization and search capabilities for dotCMS. The implementation includes a
  complete data access layer, dashboard components, and enhanced user interface for analytics data exploration.

  ## Key Features

  - Complete Analytics Dashboard: New dashboard interface with charts, metrics, and data tables
  - Advanced Data Access Layer: Dedicated analytics service with store management and data utilities
  - Enhanced Search Functionality: Improved analytics search with better query handling and routing
  - Comprehensive Testing: Full test coverage for all new components and services
  - User Agent Parsing: Browser detection and analysis utilities
  - Cube Query Builder: Advanced query construction for analytics data
  - Error Handling: Dedicated error component for analytics-related issues

###  New Libraries and Components

  - Data Access Library (libs/portlets/dot-analytics/data-access/):
    - Analytics service with comprehensive API integration
    - Dashboard store with coordinated state management
    - Type definitions for entities, cube queries, and common types
    - Utilities for browser parsing, cube queries, and data processing
  - Dashboard Components:
    - Chart visualization component with configurable displays
    - Metrics component for key performance indicators
    - Filters component for data refinement
    - Table component for detailed data views
    - Error handling component

  ### Enhanced Features

  - Routing Integration: Analytics routes properly integrated into main app routing
  - Health Guard: Analytics health check guard for route protection
  - Improved Search: Refactored search functionality with better performance
  - Localization: Updated language properties for analytics features

###  File Changes Summary

  - 63 files changed: 6,521 additions, 539 deletions
  - New libraries: Complete data access layer with services, stores, and utilities
  - Enhanced components: Dashboard, search, and error handling improvements
  - Configuration updates: Routing, project configuration, and localization

  ### Test Coverage

  - Comprehensive unit tests for all new components and services
  - 397+ test cases for analytics service
  - Extensive testing for dashboard components, utilities, and stores
  - Mock implementations for external dependencies

###  Test Plan

  - Verify analytics dashboard loads correctly
  - Test chart visualization with different data sets
  - Validate filtering and search functionality
  - Confirm error handling for network issues
  - Test responsive design across different screen sizes
  - Verify proper routing and navigation
  - Test health guard functionality

### Screenshots

https://github.com/user-attachments/assets/50d75422-39ff-47be-8bcc-2126d60ef238

This PR fixes: #32724

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -71,10 +71,15 @@ const PORTLETS_ANGULAR: Route[] = [
         loadChildren: () =>
             import('@dotcms/portlets/dot-locales/portlet').then((m) => m.DotLocalesRoutes)
     },
+    // TODO: We need a fix from BE to remove those redirects
```
Comment (zJaaal) [inline core-web/apps/dotcms-ui/src/app/app-routing.module.ts:74]: Can you create a ticket?

File: core-web/libs/portlets/dot-analytics/data-access/src/lib/services/dot-analytics.service.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        return this.#http
+            .post<AnalyticsApiResponse<UniqueVisitorsEntity>>(this.#BASE_URL, query)
+            .pipe(map((response) => response.entity[0]));
+    }
+
+    /**
+     * Top page performance metric (total requests from the most visited page)
+     */
+    topPagePerformance(
+        timeRange: TimeRange = 'from 7 days ago to now'
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-analytics/data-access/src/lib/services/dot-analytics.service.ts]: Lets create a constant for this

File: core-web/libs/portlets/dot-analytics/data-access/src/lib/store/dot-analytics-dashboard.store.ts [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+    status: ComponentStatus.INIT,
+    data: null,
+    error: null
+} as const;
+
+/**
+ * Initial store state
+ */
+const initialState: DotAnalyticsDashboardState = {
+    timeRange: 'from 7 days ago to now',
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-analytics/data-access/src/lib/store/dot-analytics-dashboard.store.ts:67]: remember to replace equal values when you have the constant

File: core-web/libs/portlets/dot-analytics/data-access/src/lib/types/common.types.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Common types for analytics data access
+ */
+
+/**
+ * Time range type for analytics queries
+ */
+export type TimeRange = 'from 7 days ago to now' | 'from 30 days ago to now' | 'this week';
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-analytics/data-access/src/lib/types/common.types.ts]: Maybe we can make this an `enum` and that way we don't need a constant value as I mentioned above

File: core-web/libs/portlets/dot-analytics/data-access/src/lib/types/cubequery.types.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+ * CubeJS query types and interfaces
+ */
+
+/**
+ * CubeJS time dimension configuration
+ */
+export interface CubeJSTimeDimension {
+    dimension: string;
+    dateRange: string | [string, string];
+    granularity?: 'day' | 'week' | 'month';
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-analytics/data-access/src/lib/types/cubequery.types.ts]: Maybe this can be an `enum`

File: core-web/libs/portlets/dot-analytics/data-access/src/lib/types/cubequery.types.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+/**
+ * CubeJS filter configuration
+ */
+export interface CubeJSFilter {
+    member: string;
+    operator: 'equals' | 'notEquals' | 'contains' | 'gt' | 'gte' | 'lt' | 'lte';
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-analytics/data-access/src/lib/types/cubequery.types.ts]: Should this be an `enum`?

File: core-web/libs/portlets/dot-analytics/data-access/src/lib/types/cubequery.types.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+/**
+ * Field types for the query builder
+ */
+export type DimensionField = 'path' | 'pageTitle' | 'userAgent' | 'createdAt' | 'eventType';
+
+export type MeasureField = 'totalRequest' | 'totalSessions' | 'totalUser';
+
+export type FilterField = 'eventType' | 'path' | 'userAgent';
+
+export type OrderField = 'totalRequest' | 'totalSessions' | 'createdAt' | 'path' | 'pageTitle';
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-analytics/data-access/src/lib/types/cubequery.types.ts]: Same question here

File: core-web/libs/portlets/dot-analytics/data-access/src/lib/utils/browser/userAgentParser.ts [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+    };
+    device: {
+        type: 'mobile' | 'tablet' | 'desktop';
+        vendor?: string;
+        model?: string;
+    };
+    os: {
+        name: string;
+        version?: string;
+    };
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-analytics/data-access/src/lib/utils/browser/userAgentParser.ts:14]: Let's create a type for every property, to prevent issues in the future and better mocking

File: core-web/libs/portlets/dot-analytics/data-access/src/lib/utils/data/analytics-data.utils.ts [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+    labels: string[];
+    datasets: {
+        label: string;
+        data: number[];
+        borderColor?: string;
+        backgroundColor?: string | string[];
+        borderWidth?: number;
+        fill?: boolean;
+        tension?: number;
+    }[];
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-analytics/data-access/src/lib/utils/data/analytics-data.utils.ts]: let's create a type for this to enhance readablitity

File: core-web/libs/portlets/dot-analytics/data-access/tsconfig.json [hunk 10]
```diff
@@ -0,0 +1,28 @@
+{
+    "compilerOptions": {
+        "target": "es2022",
+        "forceConsistentCasingInFileNames": true,
+        "strict": true,
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-analytics/data-access/tsconfig.json:5]: Check if you don't have any issues running tests

File: core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-chart/dot-analytics-dashboard-chart.component.scss [hunk 11]
```diff
@@ -0,0 +1,61 @@
+@use "variables" as *;
+
+:host {
+    display: block;
+    width: 100%;
+}
+
+.chart-card {
+    ::ng-deep .p-card-body {
+        padding: 1rem;
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-chart/dot-analytics-dashboard-chart.component.scss]: use spacing variables

File: core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-chart/dot-analytics-dashboard-chart.component.scss [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+    &--line {
+        ::ng-deep .p-skeleton {
+            border-radius: $border-radius-md;
+        }
+    }
+
+    &--pie {
+        ::ng-deep .p-skeleton {
+            width: 200px;
+            height: 200px;
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-chart/dot-analytics-dashboard-chart.component.scss]: Use rem values

File: core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-metrics/dot-analytics-dashboard-metrics.component.scss [hunk 13]
```diff
@@ -0,0 +1,51 @@
+@use "variables" as *;
+
+.metric-card {
+    ::ng-deep .p-card-body {
+        padding: 1rem;
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-metrics/dot-analytics-dashboard-metrics.component.scss]: use spacing variables

File: core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-table/dot-analytics-dashboard-table.component.html [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+                                }
+                            }
+                        </div>
+                    }
+                </div>
+            }
+
+            <!-- Skeleton Pagination -->
+            <div class="flex justify-content-center align-items-center p-3">
+                <p-skeleton height="1.5rem" width="250px"></p-skeleton>
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-table/dot-analytics-dashboard-table.component.html]: should we use rem for width too?

File: core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/constants/dot-analytics.constants.ts [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+import { FilterOption } from '../types';
+
+/** Available time period options for analytics data filtering */
+export const TIME_PERIOD_OPTIONS: FilterOption[] = [
+    { label: 'analytics.filters.time-period.1week', value: 'from 7 days ago to now' },
+    { label: 'analytics.filters.time-period.1month', value: 'from 30 days ago to now' }
+];
+
+/** Default time period selection (1 week) */
+export const DEFAULT_TIME_PERIOD: TimeRange = 'from 7 days ago to now';
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/constants/dot-analytics.constants.ts:12]: Oh, you have the constant here, check if you can reuse it or go for the enum approach

File: core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/dot-analytics-dashboard.component.scss
Comment (zJaaal) [inline core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/dot-analytics-dashboard.component.scss:1]: Check if we can change the px values for rem ones

**Review by zJaaal — DISMISSED** [review]

File: core-web/libs/portlets/dot-analytics/data-access/src/lib/services/dot-analytics.service.ts (hunk 2 above)
Comment (oidacra) [reply core-web/libs/portlets/dot-analytics/data-access/src/lib/services/dot-analytics.service.ts]: TimeRange is a `type`, only accepts that string and another...

File: core-web/libs/portlets/dot-analytics/data-access/src/lib/types/common.types.ts (hunk 4 above)
Comment (oidacra) [reply core-web/libs/portlets/dot-analytics/data-access/src/lib/types/common.types.ts]: This is the type. Doing this in this way, I can type-check the query builder. I need in that way.

File: core-web/libs/portlets/dot-analytics/data-access/tsconfig.json (hunk 10 above)
Comment (oidacra) [reply core-web/libs/portlets/dot-analytics/data-access/tsconfig.json:5]: No issues.

File: core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-chart/dot-analytics-dashboard-chart.component.scss (hunk 11 above)
Comment (oidacra) [reply core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-chart/dot-analytics-dashboard-chart.component.scss]: fixed in all components.

File: core-web/libs/portlets/dot-analytics/data-access/src/lib/utils/browser/userAgentParser.ts (hunk 8 above)
Comment (oidacra) [reply core-web/libs/portlets/dot-analytics/data-access/src/lib/utils/browser/userAgentParser.ts:14]: I created this just to show a chart. This will be deprecated when BE gives me all the parsed values.

**Review by zJaaal — APPROVED** [review]

File: core-web/libs/portlets/dot-analytics/data-access/jest.config.ts [hunk 16]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable */
+export default {
+    displayName: 'analytics-data-access',
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-analytics/data-access/jest.config.ts]: @oidacra I suggest using the prefix:  "portlet" like `portlets-dot-analytics` In this case `portlets-dot-analytics-data-access` yeah, so long, but to follow the pattern

File: core-web/libs/portlets/dot-analytics/data-access/src/lib/services/dot-analytics.service.ts [hunk 17]
```diff
@@ -0,0 +1,148 @@
...
+})
+export class DotAnalyticsService {
+    #BASE_URL = '/api/v1/analytics/content/_query/cube';
+    #http = inject(HttpClient);
+
+    /**
+     * Total de pageviews en el período especificado
+     */
+    totalPageViews(
+        timeRange: TimeRange = 'from 7 days ago to now'
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-analytics/data-access/src/lib/services/dot-analytics.service.ts]: Maybe we could use `TimeRange.LAST_7_DAYS`

File: core-web/libs/portlets/dot-analytics/data-access/src/lib/types/common.types.ts [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * Time range options for analytics filtering.
+ * Using const assertion for time range management.
+ */
+const TimeRange = {
+    LAST_7_DAYS: 'from 7 days ago to now',
+    LAST_30_DAYS: 'from 30 days ago to now',
+    THIS_WEEK: 'this week'
+} as const;
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-analytics/data-access/src/lib/types/common.types.ts:14]: 🥇

File: core-web/libs/portlets/dot-analytics/portlet/project.json [hunk 19]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "portlets-dot-analytics",
+    "name": "analytics-portlets",
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-analytics/portlet/project.json]: Other portlets follow the prefix `portlets-[my-portlet]`:

Image

File: core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-chart/dot-analytics-dashboard-chart.component.scss [hunk 20]
```diff
@@ -0,0 +1,125 @@
+@use "variables" as *;
+
+// Local variables for chart component
+$chart-pie-size: 200px;
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-chart/dot-analytics-dashboard-chart.component.scss]: Should we use rem here?

File: core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-metrics/dot-analytics-dashboard-metrics.component.scss [hunk 21]
```diff
@@ -0,0 +1,55 @@
+@use "variables" as *;
+
+.metric-card {
+    ::ng-deep .p-card-body {
+        padding: $spacing-3;
+        min-height: 114px;
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-metrics/dot-analytics-dashboard-metrics.component.scss]: use rem

**Review by nicobytes — CHANGES_REQUESTED** [review]

File: core-web/libs/portlets/dot-analytics/data-access/src/lib/store/dot-analytics-dashboard.store.ts [hunk 22]
```diff
@@ -0,0 +1,445 @@
...
+                    patchState(store, {
+                        topPagesTable: {
+                            status: ComponentStatus.LOADING,
+                            data: null,
+                            error: null
+                        }
+                    })
+                ),
+                switchMap((timeRange) =>
+                    analyticsService.getTopPagePerformanceTable(timeRange, 50).pipe(
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-analytics/data-access/src/lib/store/dot-analytics-dashboard.store.ts]: I suggest using a const for this value!

File: core-web/libs/portlets/dot-analytics/data-access/src/lib/store/dot-analytics-dashboard.store.ts [hunk 23]
```diff
@@ -0,0 +1,445 @@
...
+                                        status: ComponentStatus.LOADED,
+                                        data,
+                                        error: null
+                                    },
+                                    lastUpdated: new Date()
+                                });
+                            },
+                            (error: HttpErrorResponse) => {
+                                const errorMessage =
+                                    error.message || 'Failed to load top pages table';
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-analytics/data-access/src/lib/store/dot-analytics-dashboard.store.ts]: Should this message be translated?

File: core-web/libs/portlets/dot-analytics/data-access/src/lib/utils/browser/userAgentParser.ts [hunk 24]
```diff
@@ -0,0 +1,242 @@
...
+        name: string;
+        version: string;
+    };
+    device: {
+        type: 'mobile' | 'tablet' | 'desktop';
+        vendor?: string;
+        model?: string;
+    };
+    os: {
+        name: string;
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-analytics/data-access/src/lib/utils/browser/userAgentParser.ts:12]: Should we have a union type for this?

File: core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-chart/dot-analytics-dashboard-chart.component.html [hunk 25]
```diff
@@ -0,0 +1,85 @@
...
+            <div
+                class="flex flex-column justify-content-center align-items-center h-full text-center gap-3">
+                <i class="pi pi-exclamation-triangle text-red-500 text-2xl"></i>
+                <div class="error-message">{{ 'analytics.charts.error.loading' | dm }}</div>
+            </div>
+        </div>
+    } @else if ($isLoading()) {
+        <!-- Loading state with skeletons -->
+        <div class="chart-container" [style.height]="$height()">
+            @if ($type() === 'line') {
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-chart/dot-analytics-dashboard-chart.component.html]: Use a let template variable for type.

File: core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-filters/dot-analytics-dashboard-filters.component.scss [hunk 26]
```diff
@@ -0,0 +1 @@
+// No custom styles needed - using PrimeFlex utilities
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-filters/dot-analytics-dashboard-filters.component.scss]: If this file is not necessary, it should not be included.

File: core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-metrics/dot-analytics-dashboard-metrics.component.spec.ts [hunk 27]
```diff
@@ -0,0 +1,232 @@
...
+        component: DotAnalyticsDashboardMetricsComponent,
+        mocks: [DotMessageService]
+    });
+
+    beforeEach(() => {
+        spectator = createComponent();
+
+        // Setup the mock return value
+        const messageService = spectator.inject(DotMessageService);
+        (messageService.get as jest.Mock).mockReturnValue('Translated Message');
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/components/dot-analytics-dashboard-metrics/dot-analytics-dashboard-metrics.component.spec.ts]: This part is not necessary; just with `messageService.get..mockReturnValue('Translated Message');` is enoght

File: core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/dot-analytics-dashboard.component.scss [hunk 28]
```diff
@@ -0,0 +1,111 @@
...
+    &__dashboard-container {
+        display: flex;
+        flex-direction: column;
+        gap: $spacing-4;
+        max-width: 100%;
+    }
+
+    // Element: metrics
+    &__metrics {
+        display: flex;
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/dot-analytics-dashboard.component.scss]: We should use css class utilities

**Review by nicobytes — DISMISSED** [review]

File: core-web/libs/portlets/dot-analytics/data-access/src/lib/services/dot-analytics.service.ts (hunk 17 above)
Comment (oidacra) [reply core-web/libs/portlets/dot-analytics/data-access/src/lib/services/dot-analytics.service.ts]: I did this better
```
/**
 * Default time range for analytics queries.
 */
export const DEFAULT_TIME_RANGE = TimeRange.LAST_7_DAYS;
```

instead of exposing the const.

File: core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/dot-analytics-dashboard.component.scss (hunk 28 above)
Comment (oidacra) [reply core-web/libs/portlets/dot-analytics/portlet/src/lib/dot-analytics-dashboard/dot-analytics-dashboard.component.scss]: fixed this and others.

**Review by nicobytes — APPROVED** [review]
